Formula/azure-cli.rb

```diff
@@ -1,41 +1,496 @@
+
+  resource "xmltodict" do
+    url "https://files.pythonhosted.org/packages/57/17/a6acddc5f5993ea6eaf792b2e6c3be55e3e11f3b85206c818572585f61e1/xmltodict-0.11.0.tar.gz"
+    sha256 "8f8d7d40aa28d83f4109a7e8aa86e67a4df202d9538be40c0cb1d70da527b0df"
+  end
 
   def install
-    rm_rf "bin/windows"
+    xy = Language::Python.major_minor_version "python3"
+    site_packages = libexec/"lib/python#{xy}/site-packages"
+    ENV.prepend_create_path "PYTHONPATH", site_packages
+    ENV.prepend "LDFLAGS", "-L#{Formula["openssl"].opt_lib}"
+    ENV.prepend "CFLAGS", "-I#{Formula["openssl"].opt_include}"
+    ENV.prepend "CPPFLAGS", "-I#{Formula["openssl"].opt_include}"
+
+    # Get the CLI components we'll install
+    components = [
+      buildpath/"src/azure-cli",
+      buildpath/"src/azure-cli-core",
+      buildpath/"src/azure-cli-nspkg",
+      buildpath/"src/azure-cli-command_modules-nspkg",
+    ]
+    components += Pathname.glob(buildpath/"src/command_modules/azure-cli-*/")
+
+    # Install dependencies
+    # note: Even if in 'resources', don't include 'futures' as not needed for Python3
+    # and causes import errors. See https://github.com/agronholm/pythonfutures/issues/41
+    deps = resources.map(&:name).to_set - ["futures"]
+    deps.each do |r|
+      resource(r).stage do
+        system "python3", *Language::Python.setup_install_args(libexec)
+      end
+    end
 
-    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
-    (bash_completion/"azure").write Utils.popen_read("#{bin}/azure --completion")
+    # Install CLI
+    components.each do |item|
+      Dir.chdir(item) do
+        system "python3", *Language::Python.setup_install_args(libexec)
+      end
+    end
+
+    # This replaces the `import pkg_resources` namespace imports from upstream
+    # with empty string as the import is slow and not needed in this environment.
+    File.open(site_packages/"azure/__init__.py", "w")
+    File.open(site_packages/"azure/cli/__init__.py", "w")
+    File.open(site_packages/"azure/cli/command_modules/__init__.py", "w")
+    File.open(site_packages/"azure/mgmt/__init__.py", "w")
+
+    # TODO Find fix for https://github.com/Homebrew/brew/issues/837?
+    # Or add a Homebrew caveat that to use 'az extension' commands, 
+    # echo -e "[install]\nprefix=" > ~/.pydistutils.cfg
+    # Then you can remove this file after installing an extension.
+
+    az_exec = <<-EOS.undent
+      #!/usr/bin/env bash
+      export PYTHONPATH="#{ENV["PYTHONPATH"]}"
+      python#{xy} -m azure.cli \"$@\"
+    EOS
+    (bin/"az").write az_exec
+
+    bash_completion.install "az.completion" => "az"
+  end
 
-    # fix cxxstdlib warnings caused by installed (but not used) prebuild binaries for fibers
-    rm_rf Dir[libexec/"lib/node_modules/azure-cli/node_modules/fibers/bin/*-{46,48}"]
+  def caveats; <<-EOS.undent
+    This formula is for Azure CLI 2.0 - https://docs.microsoft.com/cli/azure/overview.
```

Comment:
This caveat isn't needed.

Reply:
Not having it could potentially confuse users.
Does Homebrew handle this automatically?

Reply:
It's really just an upgrade of version 1 to version 2 with version 1 kept around in another formula. That happens all the time, so I'm not sure why this situation requires a caveat.

Reply:
All the commands in the product have changed.
In v1, they begin with `azure ...`. In v2, they begin with `az ...`.

So the situation for confusion I see is a user upgrades then now all they `azure ...` commands no longer work.
Unless there was some kind of message on install, how would they know this?

Reply:
OK but we'll probably remove them during the next update.